### ISSUE TITLE
Allow React hook in functionnal components

### ISSUE THREAD

**Issue by GuillaumeDesforges** [jonaskello/eslint-plugin-functional#101]
It is possible to write React component as Functional Component and use "state" with React Hooks.

E.g

```typescript
const SomeComponent: React.FP<{}> = ({ }) => {
    const {state: { count }, setState} = useState<{ count: number; }>({ count: 0 });
    return (
        <p onClick={() => setState({ count: count+1 })}>{count}</p>
    )
}
```

But this triggers a "no statement" error.

**Comment by RebeccaStevens** [comment]
I'm not able to reproduce the issue you are reporting.

If you try to use `useState` without assigning the result to a variable then you would get this issue but there shouldn't be any issue with the given example.

**Comment by GuillaumeDesforges** [comment]
Then, if the callback is

```
() => {
    const something = {};
    setState(/* change state */)
}
```

It will mark as an error, but

() => {
    const something = {};
    return setState(/* change state */)
}
```

is fine ?

**Comment by RebeccaStevens** [comment]
You can use `ignorePattern` to ignore calls to `setState`.
If I remember right, this is how you can configure the rule:

```json
"functional/no-expression-statement": [
  "error", 
  {
    "ignorePattern": [
      "setState"
    ]
  }
]
```

Note that React's hook model isn't 100% functional, hence why you are noticing this issue. For React's functional components to be more functional, state would need to be passed in as a parameter and state changes would need to be returned by the function.
